smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/KafkaSourceTest.java

```diff
@@ -388,15 +506,64 @@ public void testSourceWithEmptyOptionalConfiguration() {
                 .collect(Collectors.toList())).containsExactly(0, 1, 2, 3, 4, 5, 6, 7, 8, 9);
     }
 
+    private BeanManager getBeanManager() {
+        if (container == null) {
+            Weld weld = baseWeld();
+            addConfig(new MapBasedConfig(new HashMap<>()));
+            weld.disableDiscovery();
+            container = weld.initialize();
+        }
+        return container.getBeanManager();
+    }
+
+    private Instance<KafkaConsumerRebalanceListener> getConsumerRebalanceListeners() {
+        return getBeanManager()
+                .createInstance()
+                .select(KafkaConsumerRebalanceListener.class);
+    }
+
+    private ConsumptionConsumerRebalanceListener getConsumptionConsumerRebalanceListener() {
+        return getBeanManager()
+                .createInstance()
+                .select(ConsumptionConsumerRebalanceListener.class)
+                .select(NamedLiteral.of(ConsumptionConsumerRebalanceListener.class.getSimpleName()))
+                .get();
+
+    }
+
+    private StartFromFifthOffsetFromLatestConsumerRebalanceListener getStartFromFifthOffsetFromLatestConsumerRebalanceListener(
+            String name) {
+        return getBeanManager()
+                .createInstance()
+                .select(StartFromFifthOffsetFromLatestConsumerRebalanceListener.class)
+                .select(NamedLiteral.of(name))
+                .get();
+
+    }
+
     private ConsumptionBean deploy(MapBasedConfig config) {
         Weld weld = baseWeld();
         addConfig(config);
         weld.addBeanClass(ConsumptionBean.class);
+        weld.addBeanClass(ConsumptionConsumerRebalanceListener.class);
         weld.disableDiscovery();
         container = weld.initialize();
         return container.getBeanManager().createInstance().select(ConsumptionBean.class).get();
     }
 
+    private ConsumptionBeanWithoutAck deployWithoutAck(MapBasedConfig config) {
```

Comment:
A test checking that we fail if a listener is configured but cannot be found should be added.

Reply:
done